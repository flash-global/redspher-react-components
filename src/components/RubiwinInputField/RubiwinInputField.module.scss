@import "../../css/rubiwin-constant";

.root {
    background-color: white;
    border-radius: $border-radius-sm;
    border: 1px solid $dynamiteGrey;
    padding: 2px 6px;
    transition: $transition-all;
    &:hover {
        border: 1px solid darken($dynamiteGrey, 10%);
    }
}

.focused {
    border: 1px solid $primaryColor;
    border-radius: $border-radius-sm;
    transition: $transition-all;
    &:hover {
        border: 1px solid $primaryColor;
        border-radius: $border-radius-sm;
    }
}
