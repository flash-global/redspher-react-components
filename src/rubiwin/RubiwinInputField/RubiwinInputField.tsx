import { MouseEventHandler } from 'react';
import Input from '@mui/material/Input';
import { InputProps } from '@mui/material/Input/Input';

interface IrubiwinInputField extends InputProps {
  className?: string;
  defaultValue?: any;
  disabled?: boolean;
  error?: boolean;
  endAdornment?: Node;
  fullWidth?: boolean;
  inputProps?: object;
  multiline?: boolean;
  name?: string;
  placeholder?: string;
  readOnly?: boolean;
  required?: boolean;
  rows?: number;
  startAdornment?: Node;
  type?: string;
  value?: any;
  onClick?: MouseEventHandler<HTMLDivElement>;
}

/**
 * This is a Mui Input branded for Rubiwin
 *
 * Demos:
 * - [Text Fields](https://mui.com/components/text-fields/)
 *
 * API:
 * - [Input API](https://mui.com/api/input/)
 * - inherits [InputBase API](https://mui.com/api/input-base/)
 */
const RubiwinInputField = ({
  className = '',
  id,
  defaultValue,
  disabled = false,
  endAdornment,
  error = false,
  fullWidth = false,
  inputProps,
  multiline = false,
  name,
  placeholder,
  readOnly,
  required,
  rows,
  startAdornment,
  type,
  value,
  onClick = () => '',
  ...props
}: IrubiwinInputField) => (
  <Input
    className={`${className}`}
    disableUnderline
    endAdornment={endAdornment}
    id={id}
    defaultValue={defaultValue}
    disabled={disabled}
    error={error}
    fullWidth={fullWidth}
    inputProps={inputProps}
    multiline={multiline}
    name={name}
    placeholder={placeholder}
    readOnly={readOnly}
    required={required}
    rows={rows}
    startAdornment={startAdornment}
    type={type}
    value={value}
    onClick={onClick}
    {...props}
  />
);

export default RubiwinInputField;
