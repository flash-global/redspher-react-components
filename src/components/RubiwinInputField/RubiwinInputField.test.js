import React from "react";
import renderer from "react-test-renderer";
import RubiwinInputField from "./RubiwinInputField";

const testProps = {
    className: "test",
    id: "id",
    name: "name",
    onChange: () => {},
    placeholder: "placeholder",
};

/** Snapshots */
it("renders correctly props object", () => {
    const tree = renderer.create(<RubiwinInputField {...testProps} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("is disabled", () => {
    const tree = renderer.create(<RubiwinInputField disabled />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("has error state", () => {
    const tree = renderer.create(<RubiwinInputField error />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("is full-width", () => {
    const tree = renderer.create(<RubiwinInputField fullWidth />).toJSON();
    expect(tree).toMatchSnapshot();
});
