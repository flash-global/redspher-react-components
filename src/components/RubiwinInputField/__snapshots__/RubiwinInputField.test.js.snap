// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has an input element added 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    maxLength={20}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`has error state 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root Mui-error Mui-error error"
  onClick={[Function]}
>
  <input
    aria-invalid={true}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`is disabled 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root Mui-disabled Mui-disabled disabled"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled disabled"
    disabled={true}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`is full-width 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root MuiInputBase-fullWidth MuiInput-fullWidth"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`is multiline, has 4 rows and 5 max rows 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    rows={5}
    type="text"
  />
</div>
`;

exports[`is readOnly 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    readOnly={true}
    type="text"
  />
</div>
`;

exports[`is required 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root"
  onClick={[Function]}
  required={true}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="text"
  />
</div>
`;

exports[`renders correctly props object 1`] = `
<div
  className="MuiInputBase-root MuiInput-root root test"
  onClick={[Function]}
>
  <input
    aria-invalid={false}
    className="MuiInputBase-input MuiInput-input"
    disabled={false}
    id="id"
    maxLength={20}
    name="name"
    onAnimationStart={[Function]}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="placeholder"
    type="text"
    value="rubiwin"
  />
</div>
`;
